feat: add params for initContainers, and hostNetwork

bodji opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/pyroscope?pullRequest=2363) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/pyroscope?pullRequest=2363) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/pyroscope?pullRequest=2363) it.</sub>

kolesnikovae commented on operations/pyroscope/helm/pyroscope/templates/services.yaml:
```diff
@@ -9,6 +10,10 @@ metadata:
   labels:
     {{- include "pyroscope.labels" . | nindent 4 }}
     app.kubernetes.io/component: {{ $component | quote }}
+  annotations:
+  {{- with $values.serviceAnnotations }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
```
There is an open PR https://github.com/grafana/pyroscope/pull/2364/files: we should maybe go that way. What do you think?

bodji replied:
Yes ok, I'll remove it from this one

kolesnikovae reviewed (commented):
Thank you for the contribution! 🎉 

The change looks good. I've left a couple of comments. Also, may I ask you to sign the CLA please?

bodji commented on operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml:
```diff
@@ -154,7 +161,7 @@ spec:
   {{- end }}
 {{- if eq $values.kind "StatefulSet" }}
 ---
-apiVersion: policy/v1
+apiVersion: policy/v1beta1
```
I had to change this to make it work on our v1.20.15

kolesnikovae replied:
I see, I'm sorry about that. We should probably make DisruptionBudget optional or set the api version in a smarter way:

```yaml
{{- if .Capabilities.APIVersions.Has "policy/v1" }}
apiVersion: policy/v1
{{- else }}
apiVersion: policy/v1beta1
{{- end }}
```

@bodji would you like to create another PR for this? :)

bodji commented:
I signed the CLA.

I removed serviceAnnotations (already brought by: https://github.com/grafana/pyroscope/pull/2364)

I removed the v1beta for pod disruption budget

kolesnikovae commented on operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml:
```diff
@@ -41,6 +41,13 @@ spec:
+      {{- if hasKey $values "hostNetwork" }}
+      hostNetwork: {{ $values.hostNetwork }}
+      {{- end }}
+      initContainers:
+      {{- with $values.initContainers }}
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
```
I think we should declare an empty collection in `values.yaml`: 
```yaml
initContainers: []
```

kolesnikovae replied:
```suggestion
      {{- with $values.initContainers }}
      initContainers:
      {{- toYaml . | nindent 8 }}
      {{- end }}
```

kolesnikovae reviewed (commented):
Could you also please run `make helm/check` and `make helm/docs` to update rendered manifests and the documentation

kolesnikovae commented on operations/pyroscope/helm/pyroscope/rendered/micro-services.yaml:
```diff
@@ -56,7 +56,7 @@ apiVersion: v1
 metadata:
   name: "minio-sa"
-  namespace: "default"
+  namespace: "pyroscope"
```
I think we should use the `default` namespace by default

bodji replied:
Where is this specified ? It gets changed when I do make helm/check

cyriltovena commented:
> I added a "-n default" to the helm/check helm commands in the makefile.
> 
> Because by default, it takes the current namespace set in the kube context
> 
> What do you think ?

perfect !

cyriltovena reviewed (approved):
LGTM
